cmake/vif.cmake

```diff
@@ -4,14 +4,37 @@
 # Generates USB-C VIF policies in XML format from device tree.
 set(gen_vif_script ${ZEPHYR_BASE}/scripts/generate_usb_vif/generate_vif.py)
 set(dts_compatible usb-c-connector)
-set(vif_xml ${PROJECT_BINARY_DIR}/vif.xml)
+set(vif_xml vif.xml)
+set(vif_out ${PROJECT_BINARY_DIR}/${vif_xml})
+
 set(cmd_gen_vif ${PYTHON_EXECUTABLE} ${gen_vif_script}
         --edt-pickle ${EDT_PICKLE}
         --compatible ${dts_compatible}
-        --vif-out ${vif_xml}
-        --board ${BOARD}
+        --vif-out ${vif_out}
         )
 
+if(CONFIG_GENVIF_INPUT_VIF_XML_PATH)
+    if(IS_ABSOLUTE ${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        if (EXISTS ${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+            set(vif_source_xml ${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
```

Comment:
I made this comment a month ago which is still not addressed: https://github.com/zephyrproject-rtos/zephyr/pull/52759#discussion_r1144390785
> > The CMake code can then be simplified.
>
> Code stilkl suffers from the fact that if user makes a typo, like oof.xml instead of the correct foo.xml (which is the file existing on disk), then vif_source_xml is unset and no file argument is passed to python script.
>
> Hence no build failure and user may not notice a mistake was made.

Please fix.

Reply:
The build breaks if there is a typo in the file path configured at present

As discussed offline, I added the error messages in CMAKE to say the build failed due to file not found in new revision

Reply:
I see you added an extra check below, accepted.